File: Codes/Bingran Li/String/13. Roman to Integer.java
public class Solution {
    public int romanToInt(String s) {
        // solution 1----->69%
        char [] myChar = s.toCharArray();
        int sum=0;
        for(int i=0;i<myChar.length;i++){
            if(i==myChar.length-1){ // mis "="
                sum=sum+hash(myChar[i]);
                break;
            }
            if(hash(myChar[i])<hash(myChar[i+1])){
                sum = sum-hash(myChar[i]);
            }
            else{
                sum=sum+hash(myChar[i]);
            }
        }
        
        return sum;
    }
    
    public int hash(char myChar){
        if(myChar=='M')  return 1000;
        if(myChar=='D')  return 500;
        if(myChar=='C')  return 100;
        if(myChar=='L')  return 50;
        if(myChar=='X')  return 10;
        if(myChar=='V')  return 5;
        if(myChar=='I')  return 1;
        
        return 0;
    }
    
    
}

File: Codes/Kan Zheng/C++/string/13. Roman to Integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        char level=s[0];
        int sum=0;
        for(int i=0;i<s.length();i++)
        {
            if(convert(level)<convert(s[i]))
                sum += convert(s[i])-2*convert(level);
            else{
                sum += convert(s[i]);
                level = s[i];
            }
        }
        return sum;
    }
    int convert(char r)
    {
        switch(r){
        case 'I': return 1;break;
        case 'V': return 5;break;
        case 'X': return 10;break;
        case 'L': return 50;break;
        case 'C': return 100;break;
        case 'D': return 500;break;
        case 'M': return 1000;break;
        default:return 0;break;
        }
    }
};